### PR TITLE
DEV: backport outlet wrappers

### PR DESCRIPTION
**Pull request by branquinhoaa**
- Backporting commits from main to stable.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/small-user-list.gjs [hunk 1]
```diff
@@ -38,38 +29,51 @@ export default class SmallUserList extends Component {
...
-          class="trigger-user-card"
-          data-user-card={{user.username}}
-          title={{user.username}}
-          aria-hidden="false"
-          role="listitem"
+    <PluginOutlet
+      @name="small-user-list-internal"
+      @outletArgs={{hash data=this.args}}
+    >
+      <div class="clearfix small-user-list" ...attributes>
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/small-user-list.gjs:36]: This looks like a change in HTML structure, which could affect plugins/themes. Is it absolutely necessary for the backport?

File: app/assets/javascripts/discourse/app/components/user-card-contents.hbs [hunk 2]
```diff
@@ -27,54 +27,48 @@
...
+        <PluginOutlet
+          @name="user-card-main-info"
+          @outletArgs={{hash
+            user=this.user
+            post=this.post
+            contentHidden=this.contentHidden
+            handleShowUser=this.handleShowUser
+          }}
+        >
+          <div class="user-card-avatar" aria-hidden="true">
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/user-card-contents.hbs:39]: This looks like an HTML change which may affect themes/plugins. Is it essential for the backport?

Comment (branquinhoaa) [reply app/assets/javascripts/discourse/app/components/user-card-contents.hbs:39]: it is not, I'm removing it!

File: app/assets/javascripts/discourse/app/components/small-user-list.gjs (hunk 1 above)
Comment (branquinhoaa) [reply app/assets/javascripts/discourse/app/components/small-user-list.gjs]: This div is not, the span under it contains a `role="list"` and without it I get an error `You have an element with the role of "listitem" but it is missing the required (immediate) parent element of "[group, list]` so I added this from the main branch.

File: app/assets/javascripts/discourse/app/components/small-user-list.gjs [hunk 3]
```diff
@@ -38,38 +29,49 @@ export default class SmallUserList extends Component {
...
-        >
-          {{avatar user.template "tiny"}}
-        </a>
-      {{/if}}
-    {{/each}}
+    <PluginOutlet
+      @name="small-user-list-internal"
+      @outletArgs={{hash data=this.args}}
+    >
+      <span
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/small-user-list.gjs:36]: One more html change?

File: app/assets/javascripts/discourse/app/lib/transformer/registry.js [hunk 4]
```diff
@@ -2,4 +2,10 @@ export const VALUE_TRANSFORMERS = Object.freeze([
   // use only lowercase names
   "header-notifications-avatar-size",
   "home-logo-href",
+  "home-logo-image-url",
+  "invite-simple-mode-topic",
+  "mentions-class",
+  "more-topics-tabs",
+  "post-menu-buttons",
+  "small-user-attrs",
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/lib/transformer/registry.js:10]: Some of these seem unrelated to the other changes in this PR?

File: app/assets/javascripts/discourse/app/components/small-user-list.gjs (hunk 3 above)
Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/small-user-list.gjs:36]: Oops, just saw the response you made in the other comment 👀

File: app/assets/javascripts/discourse/app/components/small-user-list.gjs (hunk 1 above)
Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/small-user-list.gjs]: Is the error from ember-template-lint? Or the browser? (if it's the former, we could just add a `{{! template-lint-ignore}}` directive

**Comment by branquinhoaa** [comment]
@davidtaylorhq I think we are good now! please let me know if I should adjust anything else.

File: app/assets/javascripts/discourse/app/components/user-card-contents.hbs (hunk 2 above)
Comment (davidtaylorhq) [reply app/assets/javascripts/discourse/app/components/user-card-contents.hbs:39]: @branquinhoaa this one is still outstanding. Is it droppable?

**Review by davidtaylorhq — APPROVED** [review]
